JFunc/src/main/java/com/jfunc/validator/FunctionValidator.java

```diff
@@ -25,92 +35,197 @@
-            if (StringUtils.equals(methodName, methodNode.name)) {
-                return getReasons(methodNode, skipLogStatements, skipPrintStatements).toString();
-            }
-        }
-        return "No method found with name" + methodName;
-    }
-
-    // At present we are checking for 1) Void 2) Print-Statements 3) Log-Statements
-    private ObjectNode getReasons(MethodNode methodNode, boolean skipLogStatements, boolean skipPrintStatements) {
-        ObjectMapper mapper = JsonUtils.getObjectMapper();
-        ObjectNode objectNode = mapper.createObjectNode();
-        MethodMetaData methodMetaData = new MethodMetaData(methodNode);
-
-        // First check whether method type is "Void"
-        if (VoidTypeUtil.isVoid(methodMetaData)) {
-            objectNode.put(JfuncConstants.ISFUNCTIONAL, false);
-            objectNode.set(JfuncConstants.REASONS, VoidTypeUtil.getReasons(methodMetaData));
-            return objectNode;
-        }
-
-        // Check for any "Print" and "Log" statements in a method
-        if (!(skipPrintStatements && skipLogStatements)) {
-            ArrayNode reasonObjectNode = mapper.createArrayNode();
-            boolean isFunctionSet = false;
-            List<InternalFeild> internalFields = methodMetaData.getInternallyRefferedFields();
-            for (InternalFeild internalField : internalFields) {
-                if (!skipPrintStatements && FieldUtil.containsPrintStatements(internalField)) {
-                    if (!isFunctionSet) {
-                        objectNode.put(JfuncConstants.ISFUNCTIONAL, false);
-                        isFunctionSet = true;
-                    }
-                    reasonObjectNode.add(FieldUtil.getReasonsForPrintStatements(internalField));
-                }
-                if (!skipLogStatements && FieldUtil.containsLogStatements(internalField)) {
-                    if (!isFunctionSet) {
-                        objectNode.put(JfuncConstants.ISFUNCTIONAL, false);
-                        isFunctionSet = true;
-                    }
-                    reasonObjectNode.add(FieldUtil.getReasonsForLogStatements(internalField));
-                }
-            }
-            if (!isFunctionSet) {
-                objectNode.put(JfuncConstants.ISFUNCTIONAL, true);
-            } else {
-                objectNode.set(JfuncConstants.REASONS, reasonObjectNode);
-            }
-        }
-        return objectNode;
-    }
+	private final String filePath;
+	private final Collection<File> classFiles;
+	private ClassReader classReader = null;
+	private ClassNode classNode;
+	private final List<FieldNode> fieldNodes = new ArrayList<FieldNode>();
+	private final AppMetadata appNode = new AppMetadata();
 
+	private List<ObjectNode> results = new ArrayList<ObjectNode>();
+
+	private final boolean skipLogStatements = false;
+	private final boolean skipPrintStatements = false;
+
+	public FunctionValidator(String path) throws FileNotFoundException {
```

Comment:
I my opinion we are not designing our project properly..

I mean it is not readable, not properly organized, not modulated, not independent, not ***, not ***,. Our project should be in a structure where even if we move out and handover it to other team they should appreciate our way of designing and implementation. Wt say guys...
:)